core/domain/collection_domain.py

```diff
@@ -435,6 +435,52 @@ def get_next_exploration_ids(self, completed_exploration_ids):
                 next_exp_ids.append(node.exploration_id)
         return next_exp_ids
 
+    def get_next_exploration_ids_in_sequence(self, current_exploration):
+        """Returns a list of exploration IDs that a logged-out user should
+        complete next based on the prerequisite skills they must have attained
+        by the time they completed the current exploration.  This recursively
+        compiles a list of 'learned skills' then, depending on the
+        'learned skills' and the current exploration's acquired skills,
+        returns either a list of exploration ids that have either just
+        unlocked or the user is qualified to explore.  If neither of these
+        lists can be generated a list of general exploration ids from the
+        collection are returned instead."""
+        collection_node = self.get_node(current_exploration)
+        learned_skills = [list(collection_node.acquired_skills)]
+        explorations_just_unlocked = []
+        explorations_qualified_for = []
+        explorations_in_general = []
+
+        def recursively_find_learned_skills(child_node):
+            """Finds the set of all skills the user must have
+            acquired by the time they've completed the current_exploration.
+            This set is then represented by learned_skills."""
+            for node in self.nodes:
+                if node != child_node:
+                    for skill in child_node.prerequisite_skills:
+                        if skill in node.acquired_skills:
+                            learned_skills[0] += node.prerequisite_skills
+                            recursively_find_learned_skills(node)
```

Comment:
maybe, to future-proof this, keep track of notes you've already recursed into, and don't recurse again if the node has already been visited?